src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const StateCopy = { ...state };
+  const res = [];
+
+  for (const key of actions) {
+    if (key.type === 'addProperties') {
+      for (const i in Object.assign(key.extraData)) {
```

Comment:
The use of `Object.assign(key.extraData)` is unnecessary here. You can directly iterate over `key.extraData` using `for (const i in key.extraData)`. `Object.assign` is typically used to merge objects, not for iteration.